mava/networks.py

```diff
@@ -0,0 +1,253 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+from typing import Dict, Sequence, Tuple, Union
+
+import chex
+import distrax
+import jax
+import jax.numpy as jnp
+import numpy as np
+from flax import linen as nn
+from flax.linen.initializers import constant, orthogonal
+
+from mava.types import (
+    Observation,
+    ObservationGlobalState,
+    RNNGlobalObservation,
+    RNNObservation,
+)
+
+
+class MLPTorso(nn.Module):
+    """MLP torso."""
+
+    layer_sizes: Sequence[int]
+    activation: str = "relu"
+    use_layer_norm: bool = False
+
+    def setup(self) -> None:
+        """Set up the activation function."""
+        if self.activation == "relu":
+            self.activation_fn = nn.relu
+        elif self.activation == "tanh":
+            self.activation_fn = nn.tanh
```

Comment:
I think this is cleaner because then we don't have to parse the string, we can just pass in the function directly

```suggestion
    activation: Callbale[[Array], Array] = nn.relu
    use_layer_norm: bool = False
```

Reply:
Or maybe to avoid setting it in all the calls of MLPTorso and to make it more flexible to control changes only  from the yaml file, we can change it to this:

```python
    activations = {"relu": nn.relu, "tanh": nn.tanh} # Add the necessary activation functions

    def setup(self) -> None:
        """Set up the activation function."""
        self.activation_fn = self.activations[self.activation]

```

Reply:
I am convinced more with @WiemKhlifi solution, any additional thoughts on that @sash-a?

Reply:
oh I didn't see the `parse_activation_fn` I think it is quite the same approach but just to make things nit I will edit based on @sash-a  suggestion :sweat_smile:

Reply:
Think that mine is shorter and lazier but sure pick what you prefer the most 😅

Reply:
Ye I don't mind either :shrug: 

I think mine is slightly more future proof so that when we make `CnnTorso`/`TransformerTorso` we don't have to redo the function

Reply:
No yours is more convincing 😅 and better for future to avoid redundancy in defining the activation_fn attribute each time 🙌